pkg/api/v1/types.go

```diff
@@ -889,3 +893,74 @@ const (
 	// CacheWriteThrough - I/O from the guest is cached on the host but written through to the physical medium.
 	CacheWriteThrough DriverCache = "writethrough"
 )
+
+// KubeVirt represents the object deploying all KubeVirt resources
+// ---
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type KubeVirt struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              KubeVirtSpec   `json:"spec,omitempty" valid:"required"`
+	Status            KubeVirtStatus `json:"status,omitempty"`
+}
+
+// Required to satisfy Object interface
+func (k *KubeVirt) GetObjectKind() schema.ObjectKind {
+	return &k.TypeMeta
+}
+
+// Required to satisfy ObjectMetaAccessor interface
+func (k *KubeVirt) GetObjectMeta() metav1.Object {
+	return &k.ObjectMeta
+}
+
+// KubeVirtList is a list of KubeVirts
+// ---
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type KubeVirtList struct {
+	metav1.TypeMeta `json:",inline"`
+	ListMeta        metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KubeVirt      `json:"items"`
+}
+
+// Required to satisfy Object interface
+func (kl *KubeVirtList) GetObjectKind() schema.ObjectKind {
+	return &kl.TypeMeta
+}
+
+// Required to satisfy ListMetaAccessor interface
+func (kl *KubeVirtList) GetListMeta() meta.List {
+	return &kl.ListMeta
+}
+
+// ---
+// +k8s:openapi-gen=true
+type KubeVirtSpec struct {
+	// The name of the VMI to perform the migration on. VMI must exist in the migration objects namespace
+	Version string `json:"version,omitempty" valid:"required"`
+}
+
+// VirtualMachineInstanceMigration reprents information pertaining to a VMI's migration.
+// ---
+// +k8s:openapi-gen=true
+type KubeVirtStatus struct {
+	Phase KubeVirtPhase `json:"phase,omitempty"`
```

Comment:
Can we immediately start with also having a conditions array here? The current phases look fine, I just want to have it visible that fine-grained status should be reflected in the conditions.